go/arrow/csv/writer.go

```diff
@@ -188,6 +188,35 @@ func (w *Writer) Write(record arrow.Record) error {
 					recs[i][j] = w.nullValue
 				}
 			}
+		case *arrow.Date32Type:
+			arr := col.(*array.Date32)
+			for i := 0; i < arr.Len(); i++ {
+				if arr.IsValid(i) {
+					recs[i][j] = arr.Value(i).FormattedString()
+				} else {
+					recs[i][j] = w.nullValue
+				}
+			}
+		case *arrow.Date64Type:
+			arr := col.(*array.Date64)
+			for i := 0; i < arr.Len(); i++ {
+				if arr.IsValid(i) {
+					recs[i][j] = arr.Value(i).FormattedString()
+				} else {
+					recs[i][j] = w.nullValue
+				}
+			}
+
+		case *arrow.TimestampType:
+			arr := col.(*array.Timestamp)
+			t := w.schema.Field(j).Type.(*arrow.TimestampType)
+			for i := 0; i < arr.Len(); i++ {
+				if arr.IsValid(i) {
+					recs[i][j] = arr.Value(i).ToTime(t.Unit).Format("2006-01-02 15:04:05.999999999")
```

Comment:
Ideally I'd have [TimestampFromStringInLocation](https://github.com/apache/arrow/blob/4bd3d2ecbb6f4f311d59098892b7dca5421c15e4/go/arrow/datatype_fixedwidth.go#L108)  but in reverse

Reply:
you can juts do `arr.Value(i).ToTime(t.Unit).In(location).Format(.....)` That's why I opted to have it use `ToTime`, because there's an `In` method on `time.Time` to easily shift it to whichever timezone location you need.